src/proto/stream.rs

```diff
@@ -201,10 +201,10 @@ where
                     self.state = StreamState::CloseAcknowledged;
                 }
             },
-            OpCode::Ping
-                if self.state == StreamState::Active
-                    && !self.frame_queue.iter().any(EncodedFrame::is_close) =>
-            {
+            // If the stream state is Active, it means none of the queued frames can be close
+            // frames, sine queue_frame would otherwise set it to ClosedByUs. That means we can skip
+            // a check here.
+            OpCode::Ping if self.state == StreamState::Active => {
```

Comment:
I've got no big preference, but switching the guard around makes this comment redundent (though I feel like it is either way)
```suggestion
            OpCode::Ping if self.state != StreamState::ClosedByUs => {
```

Reply:
I'd prefer equality checks over inequality checks since that's what clippy likes to complain about, even if it doesn't here.